### PR TITLE
More API implementations for ILGenerator 

### PR DESCRIPTION
**Pull request by buyaa-n**

### CODE REVIEW

File: src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs [hunk 1]
```diff
@@ -187,13 +229,39 @@ public override void Emit(OpCode opcode, string str)
...
+            if (opcode.Equals(OpCodes.Switch))
+            {
+                SwitchInstructionEncoder switchEncoder = _il.Switch(labels.Length);
+                UpdateStackSize(opcode);
+
+                foreach (Label label in labels)
+                {
+                    switchEncoder.Branch(_labelTable[label]);
+                }
+            }
```
Comment (jkotas) [inline src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs]:
```suggestion
            if (!opcode.Equals(OpCodes.Switch))
                throw new ArgumentException(SR.Argument_MustBeSwitchOpCode,  nameof(opcode));
                
            SwitchInstructionEncoder switchEncoder = _il.Switch(labels.Length);
            UpdateStackSize(opcode);
            
            foreach (Label label in labels)
            {
                switchEncoder.Branch(_labelTable[label]);
            }
        }
```

File: src/libraries/System.Reflection.Emit/src/Resources/Strings.resx [hunk 2]
```diff
@@ -186,4 +186,7 @@
   <data name="InvalidOperation_ShouldNotHaveMethodBody" xml:space="preserve">
     <value>Method body should not exist.</value>
   </data>
+  <data name="Argument_InvalidLabel" xml:space="preserve">
+    <value>Invalid Label.</value>
+  </data>
```
Comment (buyaa-n) [inline src/libraries/System.Reflection.Emit/src/Resources/Strings.resx:191]:
```suggestion
  </data>
  <data name="Argument_MustBeSwitchOpCode" xml:space="preserve">
    <value>Only 'OpCode.Switch' can be used.</value>
  </data>
```

File: src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs [hunk 3]
```diff
@@ -187,13 +229,40 @@ public override void Emit(OpCode opcode, string str)
...
+                
+            SwitchInstructionEncoder switchEncoder = _il.Switch(labels.Length);
+            UpdateStackSize(opcode);
+            
+            foreach (Label label in labels)
+            {
+                switchEncoder.Branch(_labelTable[label]);
+            }
+        }
+        }
```
Comment (buyaa-n) [inline src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs]:
```suggestion
        }
```

**Comment by buyaa-n** [comment]
Created a draft PR: https://github.com/dotnet/runtime/pull/93565

**Pull request closed without merging**